### PR TITLE
Drone -> GitHub Actions migration

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

Related to https://github.com/grafana/oncall-private/issues/2692

This PR simply deduplicates a lot of steps in our `linting-and-tests.yml` GitHub Actions workflow. This will make it much easier in `grafana/oncall-private` to be able to reuse some of these composable building blocks.

### CODE REVIEW

File: .github/workflows/linting-and-tests.yml [hunk 1]
```diff
@@ -263,57 +208,45 @@ jobs:
...
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: ./.github/actions/setup-python
         with:
-          python-version: "3.11.4"
-          cache: "pip"
-          cache-dependency-path: |
-            engine/requirements.txt
-            engine/requirements-dev.txt
+          python-requirements-paths: "engine/requirements.txt engine/requirements-dev.txt"
```
Comment (matiasb) [inline .github/workflows/linting-and-tests.yml:216]: If this is the default value, is it needed here? (same below)

File: .github/workflows/linting/backend-migrations.yml
Comment (matiasb) [inline .github/workflows/linting/backend-migrations.yml:1]: The plan is to move the db migrations lint job here?

**Review by matiasb — APPROVED** [review]
LGTM

File: .github/workflows/linting-and-tests.yml (hunk 1 above)
Comment (joeyorlando) [reply .github/workflows/linting-and-tests.yml]: good catch, thanks 👍

**Pull request merged**